pkg/instancetype/instancetype.go

```diff
@@ -916,6 +975,22 @@ func applyHostDevices(field *k8sfield.Path, instancetypeSpec *instancetypev1beta
 	return nil
 }
 
+func applyCPUPreferences(preferenceSpec *instancetypev1beta1.VirtualMachinePreferenceSpec, vmiSpec *virtv1.VirtualMachineInstanceSpec) {
+	if preferenceSpec.CPU == nil || len(preferenceSpec.CPU.PreferredCPUFeatures) == 0 {
+		return
+	}
+	// Only apply any preferred CPU features when the same feature has not been provided by a user already
+	cpuFeatureNames := make(map[string]struct{})
+	for _, cpuFeature := range vmiSpec.Domain.CPU.Features {
+		cpuFeatureNames[cpuFeature.Name] = struct{}{}
+	}
+	for _, preferredCPUFeature := range preferenceSpec.CPU.PreferredCPUFeatures {
+		if _, foundCPUFeature := cpuFeatureNames[preferredCPUFeature.Name]; !foundCPUFeature {
```

Comment:
What about Policy attribute comparsion? E.g. when preferences will contain e.g. vmx:require and vmi will contain vmx:forbid.

Reply:
It is only a preference so IMO it is okay to give the CPU feature in the VMI spec a higher priority.

Reply:
Yup correct, anything set in the VM/VMI overrides the preference.